metagraph/src/annotation/row_diff_builder.cpp

```diff
@@ -77,60 +78,54 @@ void build_successor(const std::string &graph_fname,
     sdsl::int_vector_buffer<1> pred_boundary(outfbase + ".pred_boundary", std::ios::out, 1024 * 1024);
     sdsl::int_vector_buffer pred(outfbase + ".pred", std::ios::out, 1024 * 1024, width);
 
-    // traverse BOSS table in parallel processing num_threads chunks of size 1'000'000
-    constexpr uint64_t chunk_size = 1'000'000;
-    const uint64_t batch_size = chunk_size * num_threads;
-    const uint64_t batch_count = (graph.num_nodes() - 1) / batch_size + 1;
-
-    ProgressBar progress_bar(batch_count, "Compute successors", std::cerr,
+    ProgressBar progress_bar(graph.num_nodes(), "Compute successors", std::cerr,
                              !common::get_verbose());
 
-    for (uint64_t batch = 0; batch < batch_count; ++batch) {
+    const uint64_t batch_size = 10'000'000;
```

Comment:
I would keep this at 1M unless you notice a speedup. I think I tried 10M and didn't notice a difference.

If you make it 10M none of the tests will be able to test the case when there is more than one batch. I believe the transcripts_1000 file has just a bit over 1M nodes.

Reply:
it was 1M * num_threads, so I made it larger. But yeah, I was hesitant to merge it yet because we don't have proper tests for this. (We also must build it using multiple threads, which is not done, I think, for transcripts_1000).

Reply:
That's true, though it was still testable, because we were only using 1 thread in transform_anno. So at least we tested the multiple batches, even if we didn't test the multithreading.
One thing people do in such cases is to add a method `set_batch_size_for_testing` that makes it clear that this is intended for testing only. We can then use this method to make the batch size 10 in unit tests. Won't work for integration tests, but maybe that's ok.

Reply:
I'll make the batch size equal to 1000, so integration tests in Debug mode will cover this case.